# Práctica Clase 2

1) Abrir el archivo "Emisiones_CO2.csv" y cargar sus datos en un diccionario.

In [41]:
import os
archivo = open('Emisiones_CO2.csv', 'r', encoding='latin-1')
print(type(archivo))

<class '_io.TextIOWrapper'>


In [42]:
dicc_emisiones = {  'cod_pais' : [],
                    'nom_pais' : [],
                    'region' : [],
                    'anio' : [],
                    'co2' : [],
                    'co2_percapita' : []}

faltantes = []
emisiones_totales = 0

lineas = archivo.readlines()
for linea in lineas:
    campos = linea.rstrip().split('|')
    dicc_emisiones['cod_pais'].append(campos[0])
    dicc_emisiones['nom_pais'].append(campos[1])
    dicc_emisiones['region'].append(campos[2])
    dicc_emisiones['anio'].append(campos[3])
    dicc_emisiones['co2'].append(campos[4])
    dicc_emisiones['co2_percapita'].append(campos[5])

    if campos[2] == 'América Latina y Caribe' and campos[3] == '2010' and campos[4] != '':
        emisiones_totales += float(campos[4].replace('.','').replace(',','.'))
        

#print(dicc_emisiones)

#Elementos faltantes
print('cod_pais', dicc_emisiones['cod_pais'].count(''))
print('nom_pais', dicc_emisiones['nom_pais'].count(''))
print('region', dicc_emisiones['region'].count(''))
print('anio', dicc_emisiones['anio'].count(''))
print('co2', dicc_emisiones['co2'].count(''))
print('co2_percapita', dicc_emisiones['co2_percapita'].count(''))

#total emisiones latam y caribe
print('\nemisiones totales latam y caribe', emisiones_totales)


cod_pais 0
nom_pais 0
region 0
anio 0
co2 2095
co2_percapita 2098

emisiones totales latam y caribe 1691634.8009999995


2) a) ¿Cuántas variables hay? <br>
b) ¿Qué tipos de datos usar para cada una? <br>
c) ¿Qué tipo de variables son? <br>
d) ¿Hay valores faltantes? <br>
e) ¿Cuál es el total de emisiones de CO2 para 'América Latina y Caribe' en el año 2010?

2.a. inicialmente hay 2 variables "archivo" y dicc_emisiones, esta última tiene dentro 6 variables de tipo lista.
2.b. para archivo, un archivo de texto. Para dicc_emisiones, se utilizan listas.
2.c. archivo es tipo TEXTWRAPPER y dicc_emisiones es un diccionario.
2.d. si, 2095 en co2 y 2098 en co2_percapita
2.e. 1691634.8009999995